src/state_migration/mod.rs

```diff
@@ -144,5 +148,70 @@ where
     Ok(None)
 }
 
+pub struct BlockstoreWithWriteBuffer<DB: Blockstore> {
```

Comment:
The code is straightforward, but it'd still be great to have some coverage there, especially given that locking mechanisms are involved.

Reply:
Done.